### ISSUE TITLE
Why does your semver minor update semantic versioning have breaking changes in minor update?

### ISSUE THREAD

**Issue by xtianus79** [angular-ui/ui-router#2505]
I haven't figured out what the exact change or bug is but a major application for my company broke because we had a ~ based on minor updates and come to find you have a major rewrite and didn't decide to at least do a version update instead of a minor bug fix update... 

I will update this when I figure out more info on the issue.

**Comment by wesleycho** [comment]
Technically not being on 1.0 means breaking changes can happen at any time according to semvar.

Also, semvar is no substitute for careful updating of libraries. A simple look at this library's commit history (and those listed in the changelog) should have also been enough to ward against this.

**Comment by robwormald** [comment]
also, rather than ranting in a totally non-actionable manner, wait until you've actually isolated the issue and then report it, so that the maintainers can actually solve the issue rather than just getting noise in their inbox.

**Comment by nateabele** [comment]
Huh, interesting. I don't recall publishing an SLA for this project. This is why we ask people to test their apps against HEAD & pre-release. If there's anything we can do to make this easier, feel free to let us know.

**Comment by xtianus79** [comment]
@robwormald @nateabele I wrote this after taking the 100 dependencies and one by one back rolling them until I figured which one was the cause.  Yes, there are preventable measures that were not taken which could have prevented this from happening.  Albeit much more difficult to do with bower.json vs package.json . 

In honest, it did get me (laughing not happily) that the thing that did us in was a "~".  Any change we make we delete and reinstall the environment so this was bound to rear its ugly head.  

When I figure out the issue I will be sure to include it here.  

Sorry for the rant... and your point of it can break regardless is well taken.
